### PR TITLE
[CompEval] test for Available errata for registered users

### PR DESCRIPTION
**Pull request by ColeHiggins2**
Test for generating host - applied errata report with registered users

### CODE REVIEW

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'
airgun: 1539

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'
airgun: 1539

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8392
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_reporttemplates.py -k test_positive_installable_errata_with_user --external-logging
Test Result : ========= 3 failed, 17 deselected, 226 warnings in 1675.12s (0:27:55) ==========
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8450
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_reporttemplates.py -k test_positive_installable_errata_with_user --external-logging
Test Result : ========= 3 failed, 17 deselected, 226 warnings in 1648.62s (0:27:28) ==========
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8459
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_reporttemplates.py -k test_positive_installable_errata_with_user --external-logging
Test Result : ========= 3 failed, 17 deselected, 229 warnings in 1645.39s (0:27:25) ==========
```

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8491
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_reporttemplates.py -k test_positive_installable_errata_with_user --external-logging
Test Result : ========= 3 passed, 17 deselected, 246 warnings in 1849.89s (0:30:49) ==========
```

File: tests/foreman/ui/test_reporttemplates.py [hunk 1]
```diff
@@ -524,3 +526,97 @@ def test_positive_generate_all_installed_packages_report(
...
+
+
+@pytest.mark.tier2
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[^6]')
+def test_positive_installable_errata_with_user(
+    session, target_sat, function_org, function_lce, function_location, rhel_contenthost
+):
+    """Generate an Installable Errata report using the Report Template - Available Errata,
+        with the option of 'Installable' with a reigistered user.
```
Comment (sambible) [inline tests/foreman/ui/test_reporttemplates.py:538]:
```suggestion
        with the option of 'Installable' with a registered user.
```

File: tests/foreman/ui/test_reporttemplates.py [hunk 2]
```diff
@@ -524,3 +526,97 @@ def test_positive_generate_all_installed_packages_report(
...
+
+    :id: 036e3ed6-b31b-4481-b591-dae438e4e61d
+
+    :setup: A Host with some applied errata
+
+    :steps:
+        1. Install an outdated package version
+        2. Apply some errata which updates the package
```
Comment (sambible) [inline tests/foreman/ui/test_reporttemplates.py:546]:
```suggestion
        2. Apply an errata which updates the package
```

**Review by sambible — CHANGES_REQUESTED** [review]
Few small docstring things, good otherwise.

**Review by vijaysawant — APPROVED** [review]
Nice work @ColeHiggins2  :clap: 
ACK

File: tests/foreman/ui/test_reporttemplates.py [hunk 3]
```diff
@@ -524,3 +526,97 @@ def test_positive_generate_all_installed_packages_report(
...
+    result = rhel_contenthost.register(
+        function_org, function_location, activation_key.name, target_sat
+    )
+    assert f'The registered system name is: {rhel_contenthost.hostname}' in result.stdout
+    assert rhel_contenthost.subscribed
+
+    # Remove package if already installed on this host
+    rhel_contenthost.execute(f'yum remove -y {FAKE_1_CUSTOM_PACKAGE_NAME}')
+    # Install the outdated package version
+    rhel_contenthost.execute(r'subscription-manager repos --enable \*')
```
Comment (vsedmik) [inline tests/foreman/ui/test_reporttemplates.py:582]: Is this needed? I believe the repo should be enabled by the AK (overriden to `enabled` already [here](https://github.com/SatelliteQE/robottelo/blob/db54ac76431e9ba52db3e6c2e3503f735176f9ef/robottelo/host_helpers/cli_factory.py#L687-L689)). 
Please correct me if I am wrong.

**Review by vsedmik — COMMENTED** [review]
Looks good to me, only one teasing question in my mind..

**Review by sambible — APPROVED** [review]

File: tests/foreman/ui/test_reporttemplates.py (hunk 3 above)
Comment (sambible) [reply tests/foreman/ui/test_reporttemplates.py:582]: seems fine to make sure in this case, no harm in the line if it ends up being unnecessary.

Comment (vijaysawant) [reply tests/foreman/ui/test_reporttemplates.py:582]: Agree with Vlad here, Repos already enabled (L564), no need to enable explicitly

Comment (ColeHiggins2) [reply tests/foreman/ui/test_reporttemplates.py]: Removed repo enabled from tr

**Comment by ColeHiggins2** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_reporttemplates.py -k 'test_positive_installable_errata_with_user'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8787
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_reporttemplates.py -k test_positive_installable_errata_with_user --external-logging
Test Result : ========= 3 passed, 17 deselected, 242 warnings in 1805.68s (0:30:05) ==========
```

**Pull request merged**